src/main/java/com/alkemy/ong/service/impl/CommentServiceImpl.java

```diff
@@ -1,8 +1,63 @@
 package com.alkemy.ong.service.impl;
 
+import com.alkemy.ong.dto.CommentDto;
+import com.alkemy.ong.exception.ParameterNotFound;
+import com.alkemy.ong.mapper.CommentMapper;
+import com.alkemy.ong.model.Comment;
+import com.alkemy.ong.repository.CommentRepository;
 import com.alkemy.ong.service.ICommentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+
 import org.springframework.stereotype.Service;
 
+
+import java.util.Locale;
+
 @Service
 public class CommentServiceImpl implements ICommentService {
+
+    @Autowired
+    private CommentRepository commentRepository;
+    @Autowired
+    private CommentMapper commentMapper;
+
+    @Autowired
+    private MessageSource messageSource;
+
+
+    @Override
+    public CommentDto save(CommentDto commentDto) {
+
+
+        try {
+            Comment commentEntity = commentMapper.commentDtoToEntity(commentDto);
+
+            if(commentEntity==null) {
+                throw new NullPointerException(messageSource.getMessage("null.dto", null, Locale.US));
+            }
+
+            try {
+                Comment savedEntity=commentRepository.save(commentEntity);
+
+
+
+                return   commentMapper.commentEntityToDto(savedEntity);
+            }
+            catch (ParameterNotFound pnf){
+
+                throw new ParameterNotFound(messageSource.getMessage("not.saved.entity",null,Locale.US));
```

Comment:
Este error no lo entiendo tampoco, por lo que veo es que está fallando el guardado y está bien que se catchee eso, pero me parece que no corresponde a un ParameterNotFound, el error es otro. Y el mensaje tampoco debería ser entidad no guardada, porque tampoco sabe el usuario que es una entidad, siempre hay que fijarse de que debe entender lo que estamos diciéndole, entonces yo podría algo como "Error al intentar guardar el comentario" por ejemplo, puede ser otra cosa, pero que sea descriptiva

Reply:
Te faltó modificar el mensaje del not.saved.entity y el tipo de exception